GliaWidgets/Coordinator/Call/CallCoordinator.swift

```diff
@@ -82,30 +86,38 @@ class CallCoordinator: SubFlowCoordinator, FlowCoordinator {
             startWith: startAction
         )
         viewModel.engagementDelegate = { [weak self] event in
-            switch event {
-            case .back:
-                self?.delegate?(.back)
-            case .engaged(operatorImageUrl: let url):
-                self?.delegate?(.engaged(operatorImageUrl: url))
-            case .finished:
-                self?.delegate?(.finished)
-            }
+            self?.handle(event: event)
         }
         viewModel.delegate = { [weak self] event in
-            switch event {
-            case .chat:
-                self?.delegate?(.chat)
-            case .minimize:
-                self?.delegate?(.minimize)
-            case .visitorOnHoldUpdated(let isOnHold):
-                self?.delegate?(.visitorOnHoldUpdated(isOnHold: isOnHold))
-            }
+            self?.handle(event: event)
         }
         return CallViewController(
             viewModel: viewModel,
             viewFactory: viewFactory
         )
     }
+
+    func handle(event: EngagementViewModel.DelegateEvent) {
```

Comment:
There's something bugging me about having two different methods called `handle` but with different events, but the thing is that it actually makes it easier to write code to handle events because you don't have to call one like `handleEngagementViewModelEvent` and another one `handleCallViewModelEvent`. So If someone else says something about it then we can think about it, otherwise just ignore me :)

Reply:
946e38598032d196fd906015b6d7e2a8e44fd974